medcat-v2/tests/utils/test_cdb_utils.py

```diff
@@ -0,0 +1,454 @@
+                },
+                average_confidence=0.8
+            )
+            cdb.cui2info["CUI1"] = cui_info1
+
+            # CUI2 without training data
+            cui_info2 = get_new_cui_info(
+                cui="CUI2",
+                preferred_name="Test Concept 2",
+                names={"test concept 2", "tc2"},
+                subnames={"test", "concept", "tc2"},
+                type_ids={"T002"},
+                description="Second test concept",
+                count_train=0,
+                context_vectors=None,
+                average_confidence=0.0
+            )
+            cdb.cui2info["CUI2"] = cui_info2
+
+            # Add name info
+            name_info1 = get_new_name_info(
+                name="test concept 1",
+                per_cui_status={"CUI1": "A"},
+                is_upper=False,
+                count_train=5
+            )
+            cdb.name2info["test concept 1"] = name_info1
+
+        elif name == "cdb2":
+            # CUI1 with different training data (should be merged)
+            cui_info1 = get_new_cui_info(
+                cui="CUI1",
+                preferred_name="Test Concept 1",
+                names={"test concept 1", "tc1", "concept one"},
+                subnames={"test", "concept", "tc1", "one"},
+                type_ids={"T001", "T003"},
+                description="First test concept (updated)",
+                count_train=15,
+                context_vectors={
+                    "long": np.random.rand(300),
+                    "medium": np.random.rand(300)
+                },
+                average_confidence=0.9
+            )
+            cdb.cui2info["CUI1"] = cui_info1
+
+            # CUI3 (new concept)
+            cui_info3 = get_new_cui_info(
+                cui="CUI3",
+                preferred_name="Test Concept 3",
+                names={"test concept 3", "tc3"},
+                subnames={"test", "concept", "tc3"},
+                type_ids={"T003"},
+                description="Third test concept",
+                count_train=5,
+                context_vectors={"short": np.random.rand(300)},
+                average_confidence=0.7
+            )
+            cdb.cui2info["CUI3"] = cui_info3
+
+            # Add name info
+            name_info1 = get_new_name_info(
+                name="test concept 1",
+                per_cui_status={"CUI1": "P"},
+                is_upper=False,
+                count_train=8
+            )
+            cdb.name2info["test concept 1"] = name_info1
+
+        # Add token counts
+        cdb.token_counts["test"] = 10
+        cdb.token_counts["concept"] = 15
+
+        return cdb
+
+    def test_merge_cdb_basic_merge(self):
+        """Test basic CDB merging functionality."""
+        merged_cdb = merge_cdb(
+            self.cdb1, self.cdb2, overwrite_training=0, full_build=True
+        )
+
+        # Should have 3 concepts total
+        self.assertEqual(len(merged_cdb.cui2info), 3)
```

Comment:
Could do this check dynamic - add together the sets of keys from both and len them for comparison.